Source/JavaScriptCore/runtime/FunctionConstructor.cpp

```diff
@@ -145,6 +145,25 @@ JSObject* constructFunction(JSGlobalObject* globalObject, const ArgList& args, c
     auto code = stringifyFunction(globalObject, args, functionName, functionConstructionMode, scope, functionConstructorParametersEndPosition);
     EXCEPTION_ASSERT(!!scope.exception() == code.isNull());
 
+    if (!args.isEmpty()) {
+        if (args.size() == 1) {
+            auto body = args.at(0);
+            RETURN_IF_EXCEPTION(scope, { });
+            if (!globalObject->globalObjectMethodTable()->canCompileStrings(globalObject, CompilationType::Function, code, args, body)) {
```

Comment:
This is too costly. `Function` constructor is actually hot function and we should not call this for the major path.
Can you instead set the information in JSGlobalObject and avoid calling out to WebCore each time we call Function constructor?

Reply:
What information could I put into JSObject, I ultimately have to be able to call code from WebCore on the arguments to Function?

Reply:
To follow up on this, the code needs to be able to execute a user defined callback for each call of the Function constructor, this callback will be unique per realm (worker and window can define their own).

One possible perf improvement is to store if the necessary CSP is configured and then only run this code when TT is present (Whether it's report only or not). That's probably the best I can do?

Once the PRs this one depends on are merged this code will be gated by a JSC option anyway but I will seak to address any perf issues regardless.

Reply:
I've updated this so it no longer calls out to WebCore *unless* you have trusted types enforced in CSP (and have the JSC option enabled)